sdk/core/azure-core/tests/test_authentication.py

```diff
@@ -230,3 +233,25 @@ def test_azure_sas_credential_policy_raises():
     sas = 1234
     with pytest.raises(TypeError):
         credential = AzureSasCredential(sas)
+
+def test_azure_named_key_credential():
+    cred = AzureNamedKeyCredential("sample_name", "samplekey")
+
+    assert cred.named_key.name == "sample_name"
+    assert cred.named_key.key == "samplekey"
+
+    cred.update("newname", "newkey")
+    assert cred.named_key.name == "newname"
+    assert cred.named_key.key == "newkey"
```

Comment:
We should have a test to make sure that we are returning a tuple as well...

Reply:
added